### PR TITLE
Prepare 3.0.0-beta.1

### PR DESCRIPTION
**Pull request by bboreham**
We've made quite a few changes.

### CODE REVIEW

**Comment by beorn7** [comment]
After the discussion that happened in https://github.com/prometheus/prometheus/pull/14958 , I think we should really try to get it into v3, because it really seems like a very annoying feature flag that people would need to enable to get the usually desired state. At is _could_ create surprising effects, it would be good have it in a beta release, and so I would ask to have it in this beta.1 (so that we keep open the option of not having a beta.2…).

**Comment by bboreham** [comment]
To close out an earlier discussion, the PR referred as #14958 was merged as:
 * #14160

**Comment by beorn7** [comment]
WRT the default scrape port: As discussed elsewhere, the current state in main attempts to implement the actually desired behavior, which is a bit different from both previous states. To summarize:

- old state without feature flag: add default port (80 or 443) if no port specified by the user
- old state with feature flag: don't add the default port if not specified by the user but _remove_ it if specified by the user
- new state: take exactly what the user specified (which is presumable the behavior of every normal HTTP library)

File: CHANGELOG.md
Comment (beorn7) [inline CHANGELOG.md:9]: Maybe be more explicit and say that now Prometheus will never add or remove the default port.

File: CHANGELOG.md
Comment (beorn7) [inline CHANGELOG.md:15]: Nit: Period is doubled here.

**Review by beorn7 — CHANGES_REQUESTED** [review]

**Review by ArthurSens — APPROVED** [review]

File: CHANGELOG.md
Comment (beorn7) [inline CHANGELOG.md]: Now we can add #15096 #15082 here.

**Comment by bboreham** [comment]
I have moved this branch forward to latest commit on main as of a couple of hours ago (90f7832447d).
And updated per @beorn7's review.

File: CHANGELOG.md
Comment (beorn7) [inline CHANGELOG.md:5]: Let's add the new stuff that have landed in main by now. Maybe incomplete list (and not sure if tagging all the "declare stable" items as ENHANCEMENTs is the right thing):

- [FEATURE] SD: Support K8s sidecar containers in endpoint discovery. #14929
- [ENHANCEMENT](?) Declare OTLP receiver stable. #14894
- [ENHANCEMENT](?) Declare "float literals are time durations" as stable. #15111
- [BUGFIX] Re-add accidentally removed flag storage.tsdb.allow-overlapping-compaction. #15113
- [BUGFIX] storage: Fix missing histogram copy in sampleRing. #15025
- [BUGFIX] scrape: Fix protobuf parsing of unit field. #15095
- [BUGFIX] Fixes and optimisation for created timestamp ingestion. #14965

**Review by beorn7 — CHANGES_REQUESTED** [review]

**Comment by beorn7** [comment]
I'm checking my proposed changes against yours.

File: CHANGELOG.md
Comment (beorn7) [inline CHANGELOG.md:12]: I guess `* [FEATURE] SD: Support K8s sidecar containers in endpoint discovery. #14929` is still missing here?

File: CHANGELOG.md
Comment (beorn7) [inline CHANGELOG.md:19]: Not sure if an ENHANCEMENT or something else, but we should probably mention `Float literals and time durations being the same is now a stable fetaure. #15111`

File: CHANGELOG.md
Comment (beorn7) [inline CHANGELOG.md:30]:
```
* [BUGFIX] Re-add accidentally removed flag storage.tsdb.allow-overlapping-compaction. #15113
* [BUGFIX] Fixes and optimisations for created timestamp ingestion. #14965
```

**Review by beorn7 — CHANGES_REQUESTED** [review]
OK, I think these are the remaining diffs between what I found and what you found.

File: CHANGELOG.md
Comment (bboreham) [reply CHANGELOG.md:5]: #14965 did not look user-visible. 
#15095 is in my text. 
#15025 is in my text.
#15113 I omitted deliberately and asked about on the issue.
#15111 was marked 'docs' so omitted from changelog.
#14894 was already in CHANGELOG so not changed.
#14929 does not appear in my git log

File: CHANGELOG.md
Comment (bboreham) [reply CHANGELOG.md:30]: https://github.com/prometheus/prometheus/pull/14965 did not look user-visible.
https://github.com/prometheus/prometheus/pull/15113 I omitted deliberately and asked about on the issue.

Comment (beorn7) [reply CHANGELOG.md:30]: It looks you are right about #14965. I thought it included the fix of not including the separate _created time series, but apparently that was in a previous PR.

#15113 is an edge case. It's a hidden flag, but even a hidden flag is part of the user interface. See update on the issue. But ultimately your call. I'm fine with either.

File: CHANGELOG.md
Comment (bboreham) [reply CHANGELOG.md:12]: The commits from that PR do not mention "#14929" so it was missed.
I have asked about this on the issue.

Interested to know how you picked it up?

File: CHANGELOG.md
Comment (beorn7) [reply CHANGELOG.md:19]: The change is indeed just in the doc, but maybe still important enough. (As in: If this had been guarded by a feature flag, we would now have removed it.)

File: CHANGELOG.md
Comment (bboreham) [reply CHANGELOG.md:30]: #14965 should be back-ported to release-2.55 and then it can go on the same line as #14356.

File: CHANGELOG.md
Comment (bboreham) [reply CHANGELOG.md]: I put #15082 on a new line, as notifications seemed to be a notable enhancement. (is it a feature?)
Are there any other kinds of UI notifications?

File: CHANGELOG.md
Comment (beorn7) [reply CHANGELOG.md]: I just saw the commit itself (and it was one of those where the commit description was the unedited result of the squashing :cry: )

**Review by beorn7 — APPROVED** [review]
Approving now. (You could wait for UI experts to chime in, but I guess with the UI being completely new in v3, we don't really have to spell out every single feature.)

**Comment by juliusv** [comment]
> Approving now. (You could wait for UI experts to chime in, but I guess with the UI being completely new in v3, we don't really have to spell out every single feature.)

Agreed :+1:

**Review by ArthurSens — APPROVED** [review]

**Pull request merged**
